### ISSUE TITLE
[caffe2] How to handle multiple inputs and multiple outputs in the network architecture?

### ISSUE THREAD

**Issue by CarlosYeverino** [pytorch/pytorch#13041]
Is it currently possible to handle multiple inputs and multiple outputs in the network architecture in CAFFE2?

**Comment by BIGBALLON** [comment]
Hi, @CarlosYeverino 

It's easy to handle multiple inputs and multiple outputs.
Just add the blobs to workspace. Then add some operation to calculate the loss .
After inference, you can get the blob by name, then use it in your project.

**Comment by CarlosYeverino** [comment]
Hi @BIGBALLON ,

thanks for your replay!

How would you handle the input feeding in training if you use brew.db_input? Would you call the add_input function for each desired input?

```
def add_input(self, model, batch_size, db, db_type, device_opts):
        with core.DeviceScope(device_opts):
            # load the data
            data_uint8, label = brew.db_input(
                model,
                blobs_out=["data_uint8", "label"],
                batch_size=batch_size,
                db=db,
                db_type=db_type,
            )
            # cast the data to float
            data = model.Cast(data_uint8, "data", to=core.DataType.FLOAT)

            # scale data from [0,255] down to [0,1]
            data = model.Scale(data, data, scale=float(1./256))

            # don't need the gradient for the backward pass
            data = model.StopGradient(data, data)
            return data, label
```
